torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1389,18 +1387,14 @@ void ProcessGroupNCCL::heartbeatMonitor() {
       if (heartbeat != heartBeatCounter) {
         heartBeatCounter = heartbeat;
       } else {
-        if (!shouldDump_.load()) {
-          LOG(ERROR)
-              << logPrefix()
-              << "First PG on this rank that detected no heartbeat of its watchdog.";
-        }
         shouldDump_.store(true);
-        // No heartbeat increase detected and timeout.
+        // No watchdog heartbeat increase detected and timeout.
         errorMsg = c10::str(
             logPrefix(),
-            "Heartbeat monitor timed out! Process will be terminated after dumping debug info.",
-            " workMetaList_.size()=",
-            workMetaList_.size());
+            "No heartbeat detected for watchdog thread, ",
```

Comment:
"No heartbeat detected for watchdog thread.  This usually indicates calls into CUDA apis (e.g. CudaEventDestroy) are hanging, and the root cause may be outside of distributed communication operations."

Reply:
I moved the exit msg to the error msg.